### PR TITLE
regions plugin: display regions before file load

### PR DESCRIPTION
**Pull request by TimSC**
Pre-generated peaks waits for the audio length, either from loaded audio or from explicitDuration. This PR adds explicitDuration as an argument for the WebAudio constructor. This allows it to be defined up front.

I've also updated the regions plugin initialization to make use of this information so the "backend-created" event is used to add regions, rather than the "ready" event which only fires when a song has finished loading.

### CODE REVIEW

**Comment by thijstriemstra** [comment]
thanks @TimSC this looks fine. can you also add a unit test in `spec/wavesurfer.spec.js` for this?

**Comment by TimSC** [comment]
I will try but I'm not that great at javascript!

File: src/plugin/regions.js [hunk 1]
```diff
@@ -177,31 +177,34 @@ class Region {
...
     updateRender() {
+        // Duration varies during loading process, so don't overwrite important data
         const dur = this.wavesurfer.getDuration();
         const width = this.getWidth();
 
-        if (this.start < 0) {
-            this.start = 0;
-            this.end = this.end - this.start;
+        var start_limited = this.start;
+        var end_limited = this.end;
```
Comment (katspaugh) [inline src/plugin/regions.js]: Could you write these variables with camelCase?

File: src/wavesurfer.js [hunk 2]
```diff
@@ -187,6 +187,7 @@ export default class WaveSurfer extends util.Observer {
         cursorColor: '#333',
         cursorWidth: 1,
         dragSelection: true,
+        explicitDuration: null,
```
Comment (katspaugh) [inline src/wavesurfer.js]: I would call it simply `duration` and mark as optional in the docs.

**Comment by thijstriemstra** [comment]
ping @TimSC

**Comment by TimSC** [comment]
Great thanks so much! I was a bit out of my depth using javascript.

**Pull request merged**